### PR TITLE
Move to @wordpress/escape-html and @wordpress/date packages

### PR DESCRIPTION
**Pull request by erikyo**
## What?
Initially (out of habit) I had used the Moment and domPurify packages, but similar things already exist in the Gutenberg repo and it is definitely better to use these, since now the compiled sources weigh a tenth as much as before.

## How?
I make amends and this PR updates to WP built in packages

### CODE REVIEW

File: src/scripts/components/NoticeMeta.js [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
- * @param {number} props.date   The date of the notification.
+ * @param {string} props.date   The date of the notification.
  * @param {string} props.source The source of the notification.
  */
 export const NoticeMeta = ( { date, source } ) => (
 	<p className="wp-notification-meta">
 		<span className="name">{ source }</span> { '\u2022 ' }
-		<span className="date">{ moment( date ).fromNow() }</span>
+		<span className="date">
+			{ dateI18n( 'l jS F Y - h:i A', date, true ) }
```
Comment (johnhooks) [inline src/scripts/components/NoticeMeta.js]: Can you make a utility function that does this named `fromNow` or `timeFromNow`?

Comment (erikyo) [reply src/scripts/components/NoticeMeta.js]: Actually just converts the epoch to a readable time... Would it be better `toCurrentDate`?

File: src/scripts/components/NoticeMeta.js [hunk 2]
```diff
@@ -1,15 +1,15 @@
-import * as moment from 'moment/moment';
+import { noticeDateFormat } from '../utils';
 
 /**
  * The notice metadata, for example the source of the notification or the date
  *
  * @param {Object} props
- * @param {number} props.date   The date of the notification.
+ * @param {string} props.date   The date of the notification.
```
Comment (johnhooks) [inline src/scripts/components/NoticeMeta.js]: The date is a string?

Comment (erikyo) [reply src/scripts/components/NoticeMeta.js]: seems the [dateI18n accepts](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-date/#datei18n) only stings and not numbers

**Review by johnhooks — APPROVED** [review]
I think these are great changes and really reduce the size of this package and uses the WordPress provided functionality.

**Pull request merged**